docs/source/reference/tpu.rst

```diff
@@ -0,0 +1,168 @@
+.. _tpu:
+
+Cloud TPU
+================================
+
+SkyPilot supports running jobs on `Cloud TPU <https://cloud.google.com/tpu/docs/intro-to-tpu>`_
```

Comment:
Consider front loading snippets somewhere here. Example text:

tl;dr: Get a TPU Node with one command:
```
sky tpunode
sky tpunode --use-spot
sky tpunode --tpus tpu-v2-8
```
For TPU VMs, see below.

---

Should we ban the following?
```
» sky tpunode --tpus tpu-v3-32                                                                  1 ↵
I 08-23 09:46:02 resources.py:238] Missing runtime_version in accelerator_args, using default (2.5.0)
I 08-23 09:46:03 optimizer.py:605] == Optimizer ==
I 08-23 09:46:03 optimizer.py:628] Estimated cost: $32.5 / hour
I 08-23 09:46:03 optimizer.py:628]
I 08-23 09:46:03 optimizer.py:675] Considered resources (1 node):
I 08-23 09:46:03 optimizer.py:703] -----------------------------------------------------------
I 08-23 09:46:03 optimizer.py:703]  CLOUD   INSTANCE       ACCELERATORS   COST ($)   CHOSEN
I 08-23 09:46:03 optimizer.py:703] -----------------------------------------------------------
I 08-23 09:46:03 optimizer.py:703]  GCP     n1-highmem-8   tpu-v3-32:1    32.47         ✔
I 08-23 09:46:03 optimizer.py:703] -----------------------------------------------------------
I 08-23 09:46:03 optimizer.py:703]
Launching a new cluster 'sky-tpunode-zongheng'. Proceed? [Y/n]: n
Aborted!
```

Reply:
I hesitated a bit because GCP seems to suggest TPU VM as the default option and is retiring TPU Nodes. For example, you can't easily find TPU Node tutorial on their documentation. We may want to set TPU VM as default for our tpunode.

We do support `sky tpunode --tpus tpu-v3-32` now (because it's TPU Node), which Sehoon is currently using.